pkg/workload/workload.go

```diff
@@ -1302,6 +1315,14 @@ func WithCustomPrepare(customPrepare func(wl *kueue.Workload)) EvictOption {
 	}
 }
 
+// EvictWithRetryOnConflict sets the RetryOnConflict field in EvictOptions to true, enabling retrials on conflict during eviction.
+// Note: This only works with merge patches.
+func EvictWithRetryOnConflict() EvictOption {
```

Comment:
```suggestion
func WithRetryOnConflictForPatch() EvictOption {
```

Reply:
Yes, but we are using Evict on scheduler as well. https://github.com/kubernetes-sigs/kueue/pull/7600#discussion_r2560570400.

Reply:
Yes, but it is using strict mode, so for now I don't see a need to introduce the RetryOnConflict. 

We can consider in the future. For now I see the goal of this PR to copy only to `LooseOnApply=true, LooseOnPatch=false.`. I think we could even throw "unsupported" error if the code tries to use RetryOnConlict in other scenario.

Reply:
OK. Let me revert it.